core/error.go

```diff
@@ -177,3 +177,20 @@ func IsValueTransferErr(e error) bool {
 	_, ok := e.(*ValueTransferError)
 	return ok
 }
+
+type BadHashErr struct {
+	Message string
+}
```

Comment:
Nitpicking, but the stdlib way of defininf custom errors ends if `Error` fully written out:

 * http://golang.org/pkg/net/#DNSError
 * http://golang.org/pkg/net/http/#ProtocolError

I think it would be better to keep to that convention.